Simplified/Audiobooks/LCP/LCPAudiobooks.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  LCPAudiobooks.swift
+//  Simplified
+//
+//  Created by Vladimir Fedorov on 16.11.2020.
+//  Copyright © 2020 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+import R2Shared
+import R2Streamer
+import NYPLAudiobookToolkit
+
+/// LCP Audiobooks helper class
+@objc class LCPAudiobooks: NSObject {
+    
+  private let audiobookUrl: URL
+  private let lcpService = LCPLibraryService()
+
+  /// LCP license object for decryption
+  private var lcpLicense: DRMLicense?
+
+  /// .lcpa archive container
+  private let container: Container
+
+  /// Distributor key - one can be found in `NYPLBook.distributor` property
+  @objc static let distributorKey = "lcp"
+  
+  /// Initialize for an LCP audiobook
+  /// - Parameter audiobookUrl: must be a file with `.lcpa` extension
+  @objc init?(for audiobookUrl: URL) {
+    self.audiobookUrl = audiobookUrl
+    do {
+      guard let (pubBox, _) = try Publication.parse(at: self.audiobookUrl) else {
+        return nil
+      }
+      self.container = pubBox.associatedContainer
+    } catch {
+      NYPLErrorLogger.logError(error, summary: "Error opening audiobook")
+      return nil
+    }
+  }
+  
+  /// Content dictionary for `AudiobookFactory`
+  @objc func contentDictionary() -> NSDictionary? {
+    let manifestPath = "manifest.json"
+    do {
+      // Relative path inside the audiobook
+      let data = try container.data(relativePath: manifestPath)
+      let publicationObject = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as? NSDictionary
+      return publicationObject
+    } catch {
+      NYPLErrorLogger.logError(error, summary: "Error reading audiobook \(manifestPath)")
+      return nil
+    }
+  }
+
+}
+
+/// DRM Decryptor for LCP audiobooks
+extension LCPAudiobooks: DRMDecryptor {
+
+  /// Decrypt protected file
+  /// - Parameters:
+  ///   - url: encrypted file URL.
+  ///   - resultUrl: URL to save decrypted file at.
+  ///   - completion: decryptor callback with optional `Error`.
+  func decrypt(url: URL, to resultUrl: URL, completion: @escaping (Error?) -> Void) {
+    loadLicense { error in
+      if let error = error {
+        completion(error)
```

Comment:
is this error reported to NYPLErrorLogger on the calling side? if not, we should do that here (assuming here it's better than there, I don't know about that). After that, it would be good to document what logging this method is doing too.

Reply:
Yes, it is in `NYPLAudiobookToolkit` https://github.com/NYPL-Simplified/NYPLAudiobookToolkit/blob/master/NYPLAudiobookToolkit/Player/LCPPlayer.swift line 33:

```
            delegate.decrypt(url: task.url, to: decryptedUrl) { error in
                if let error = error {
                    ATLog(.error, "Error decrypting file", error: error)
                    return
                }
```

Reply:
I see, that's pretty low level and it's just a log line. I would file an actual report with `NYPLErrorLogger.logError` so these surface on crashlytics. Make sure to mention LCP in the summary.